.github/workflows/e2e.yaml

```diff
@@ -97,10 +104,18 @@ jobs:
       - id: generate-cluster-name
         name: generate cluster name
         run: |
-          CLUSTER_NAME="$(echo ${{ inputs.suite }}-"$RANDOM$RANDOM" | awk '{print tolower($0)}' | tr / -)"
-          echo Using cluster name "$CLUSTER_NAME"
+          CLUSTER_NAME='${{ inputs.cluster_name }}'
+          if [[ '${{ inputs.cluster_name }}' == '' ]]; then 
+            if [[ '${{ inputs.workflow_trigger }}' == 'soak' ]]; then 
+              CLUSTER_NAME=$(echo soak-periodic-$RANDOM$RANDOM | awk '{print tolower($0)}' | tr / -)
+            else 
+              CLUSTER_NAME=$(echo ${{ inputs.suite }}-$RANDOM$RANDOM | awk '{print tolower($0)}' | tr / -)
+            fi 
+          fi
+          echo "Using cluster name \"$CLUSTER_NAME\""
           echo CLUSTER_NAME="$CLUSTER_NAME" >> "$GITHUB_OUTPUT"
       - name: setup eks cluster '${{ steps.generate-cluster-name.outputs.CLUSTER_NAME }}'
+        if: inputs.cluster_name == ''
```

Comment:
Looks like e2e/setup-cluster assumes that cluster name is set. When do we actually generate the cluster name? 

Required: https://github.com/aws/karpenter/blob/main/.github/actions/e2e/setup-cluster/action.yaml#L25-L27
Used as input to a function: https://github.com/aws/karpenter/blob/main/.github/actions/e2e/setup-cluster/action.yaml#L66
same pt2: https://github.com/aws/karpenter/blob/main/.github/actions/e2e/setup-cluster/action.yaml#L83